# cancer-module-sv
Bioinformatics Training Platform (BTP) Module: Structural Variant Analysis

  * Topic
    * Structural Variant Analysis
  * Target Audience
    * Biologists
	* Non-bioinformaticians
	* Little to no programming expereience
  * Prerequisites
    * None
  * Key Learning Outcomes
    * Have been provided with key fundamentals on how paired-end mappings and split-read/soft-clipped read patterns are used in detecting deletions, tandem duplicates, inversions and translocations. 
    * Know what important quality control checks need to be evaluated prior to structural variant calling. 
    * Have run DELLY on a subset of whole genome next generation sequencing data pertaining to a single human tumour with a matched normal control.
    * Be able to filter high confidence SV predictions. 
    * Have gained basic knowledge to interpret the VCF output provided by DELLY.
    * Have used their understanding of distinct SV paired-end mapping and soft-clipped read patterns to visually verify DELLY predicted SVs using IGV.
  * Time Required
    * 

License
=======
The contents of this repository are released under the Creative Commons
Attribution 3.0 Unported License. For a summary of what this means,
please see:
http://creativecommons.org/licenses/by/3.0/deed.en_GB
